runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpUtils.kt

```diff
@@ -88,6 +80,19 @@ internal fun HttpRequest.toOkHttpRequest(
     return builder.build()
 }
 
+private fun Headers.toOkHttpHeaders(): OkHttpHeaders = OkHttpHeaders.Builder().also { okHeaders ->
+    forEach { key, values ->
+        values.forEach { value ->
+            okHeaders.addUnsafeNonAscii(key, value)
+        }
+    }
+
+    if ("Accept-Encoding" !in this) {
+        // Disable OkHttp transparent response decompression. See https://github.com/awslabs/smithy-kotlin/issues/1041
+        okHeaders.addUnsafeNonAscii("Accept-Encoding", "identity")
```

Comment:
**Question**: This doesn't cause issues with signatures? I didn't see anywhere in our canonicalizer or CRT where `Accept-Encoding` is ignored.

Reply:
I think it's not an issue because `toOkHttpRequest` is called after canonicalization / signing, and `Accept-Encoding` is not sent as a signed header

Reply:
Oh ok, if you add `Accept-Encoding` before signing (like in an interceptor), yes it does end up as a signed header. All headers are signed by default unless someone configures `shouldSignHeader` predicate.

Reply:
Right so that would result in a signature mismatch server side I would assume.

Reply:
I tried adding an `Accept-Encoding: gzip` header _before_ signing for a request to S3 and it succeeded, can't say that will work for all services though.

Reply:
If some other feature/interceptor added an `Accept-Encoding` header, it would go through our regular signing logic and be included in the signature calculation. Because this PR adds the header post-signing _only if it doesn't already exist_, it will be unsigned.

According to [the public SigV4 spec](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html), `Accept-Encoding` is not _required_ to be signed:
> The `CanonicalHeaders` list must include the following:
> * HTTP `host` header.
> * If the `Content-Type` header is present in the request, you must add it to the `CanonicalHeaders` list.
> * Any `x-amz-*` headers that you plan to include in your request must also be added. For example, if you are using temporary security credentials, you need to include `x-amz-security-token` in your request. You must add this header in the list of `CanonicalHeaders`.

That wording implies then that any additional headers in the request _may_ be canonicalized and included in the signature but are not required to be. (The `SignedHeaders` component of the `Authorization` header would definitively identify which headers were included in the signature calculation.)

That said, it's possible that some service-side implementations of SigV4 verification may have different assumptions about which headers must be signed. I've run this PR against our (admittedly small) suite of E2E tests against Kinesis, Polly, S3, STS, Timestream, and Transcribe, all of which pass.

Reply:
FWIW, I just ran our service benchmark suite as well, which touches S3, SNS, STS, CloudWatch, CloudWatch Events, DynamoDB, Pinpoint, and IAM. Again, all calls succeed.

That the E2E tests and benchmarks all pass indicates to me that _these_ services, at least, do not require `Accept-Encoding` to be signed (which is in agreement with the SigV4 spec).

Reply:
👍 Cool I just wanted to confirm we'd verified this.

> The SignedHeaders component of the Authorization header would definitively identify which headers were included in the signature calculation.

Yes but I know in the past I want to say we still hit issues where services didn't seem to care. This all looks fine though and we've done due diligence.